src/rpdk/core/project.py

```diff
@@ -217,15 +221,37 @@ def generate(self):
         # generate template for IAM role assumed by cloudformation
         # to provision resources if schema has handlers defined
         if "handlers" in self.schema:
+            handlers = self.schema["handlers"]
             template = self.env.get_template("resource-role.yml")
             permission = "Allow"
             path = self.root / ROLE_TEMPLATE_FILENAME
             LOG.debug("Writing Resource Role CloudFormation template: %s", path)
             actions = {
                 action
-                for handler in self.schema["handlers"].values()
+                for handler in handlers.values()
                 for action in handler.get("permissions", [])
             }
+
+            # calculate IAM role max session timeout based on highest handler timeout
+            # with some buffer (70 seconds per minute)
+            cud_handlers = [
+                handlers.get(handler, {})
+                for handler in ["create", "update", "delete"]
+                if handlers.get(handler, None)
+            ]
+            max_handler_timeout = max(
+                (
+                    handler.get("timeoutInMinutes", DEFAULT_ROLE_TIMEOUT_MINUTES)
+                    for handler in cud_handlers
+                ),
+                default=DEFAULT_ROLE_TIMEOUT_MINUTES,
+            )
+            # max role session timeout must be between 15 minutes and 12 hours
+            role_session_timeout = min(
```

Comment:
I would suggest to validate and give error if the `timeoutInMinutes` configuration is not between 15 ~ 12*60.  Translate the invalid values to a min value and max value would confuse customer thought their configuration accepted, however, the actual value is not as same as configured.

Reply:
this value will be validated by the schema itself so I don't think that's necessary, and that includes with the buffer we give (multiplication by 70)

Reply:
also the schema allows the timeout to be 2 minutes, so we don't want to validate that.  Actually the max role session timeout [must be between 1 hour and 12 hours](https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateRole.html) so I will have to update this